src/main/java/com/arpnetworking/tsdcore/sinks/AggregationServerHttpSink.java

```diff
@@ -47,20 +57,78 @@ protected String getContentType() {
     }
 
     @Override
-    protected Collection<byte[]> serialize(final PeriodicData periodicData) {
+    protected Collection<SerializedDatum> serialize(final PeriodicData periodicData) {
         final Collection<Messages.StatisticSetRecord> records = PeriodicDataToProtoConverter.convert(periodicData);
 
-        final ImmutableList.Builder<byte[]> serializedPeriodicData = ImmutableList.builder();
+        final ImmutableList.Builder<SerializedDatum> serializedPeriodicData = ImmutableList.builder();
         for (final Messages.StatisticSetRecord record : records) {
-            serializedPeriodicData.add(AggregationMessage.create(record).serializeToByteString().toArray());
+            serializedPeriodicData.add(new SerializedDatum(
+                    AggregationMessage.create(record).serializeToByteString().toArray(),
+                    computePopulationSize(record.getMetric(), record.getStatisticsList())
+                    )
+            );
         }
         return serializedPeriodicData.build();
     }
 
+    private static long computePopulationSize(final String metricName, final List<Messages.StatisticRecord> statisticsList) {
```

Comment:
So I dont think this is how we want to do it. Based on our previous conversation the assumption was that we could obtain this information from the Protobuf structures. However, it's clear that this is not possible because we have to deserialize the supporting data. 

I think it would be better to push the logic down into `PeriodicDataToProtoConverter` and have it return a structure which contained the population size with each `Messages.StatisticSetRecord`.